File: ResourceManager/Models/Permission.cs
using ResourceManager.Models.Abstractions;
using System.Collections.Generic;

namespace ResourceManager.Models
{
    public class Permission : BaseEntity, IItem
    {
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; }= string.Empty;
        public virtual IList<RolePermission> RolePermissions { get; set; }  = new List<RolePermission>();
        public override string GetCaption()
        {
            return string.Format("{0} with name '{1}'",nameof(Permission),  Name);
        }
    }
}

File: ResourceManager/Models/Permission.vb
Imports ResourceManager.Models.Abstractions
Imports System.Collections.Generic

Namespace ResourceManager.Models
    Public Class Permission
        Inherits BaseEntity
        Implements IItem
        Public Property Name As String = String.Empty Implements IItem.Name
        Public Property Description As String = String.Empty
        Public Overridable Property RolePermissions As ICollection(Of RolePermission)
        Public Overrides Function GetCaption() As String
            Return String.Format("{0} with name '{1}'", NameOf(Permission), Name)
        End Function
    End Class
End Namespace
